lib/mix/lib/mix/compilers/elixir.ex

```diff
@@ -120,6 +120,7 @@ defmodule Mix.Compilers.Elixir do
         {:ok, warning_diagnostics(sources)}
 
       true ->
+        write_manifest(manifest, [], [], %{}, timestamp)
```

Comment:
Actually, you were right. If we always do this, it means we will always trigger downstream compilers. :D Sorry!

```suggestion
        if not File.exists?(manifest) do
          write_manifest(manifest, [], [], %{}, timestamp)
        end
```

Reply:
do you prefer this or changing it back to being part of the cond?

Reply:
This is more appropriate because we still want to return no-op. :)